react/SquareAppIcon/index.jsx

```diff
@@ -61,9 +62,10 @@ const useStyles = makeStyles(theme => ({
   }
 }))
 
-export const SquareAppIcon = ({ app, type, name, variant, IconContent }) => {
+export const SquareAppIcon = ({ app, type, name, variant, IconContent, appIconProps = {} }) => {
+  // "app" and "type" properties are deprecated in favor of appIconProps
   const classes = useStyles()
-  const appName = name || (app && app.name) || app || ''
+  const appName = name || (get(appIconProps, 'app.name')) || (app && app.name) || app || ''
```

Comment:
why don't we do the same for appSlug? 

If I read the code correctly, we style need to use `app` props to specify the slug right? 

So with that change https://github.com/cozy/cozy-ui/pull/1973/files#diff-7bbce49ffc2d1b4a041913caac7d27c2d1812a321158075dc1094df8978459b6R21 we pass an `undefined` to <AppIcon app={undefined}` right?

Reply:
The app slug is only used by AppIcon. I don't understand why we would need to specify the slug.

> we pass an undefined to <AppIcon app={undefined} right?

    app = { name: "Test App", slug: "testapp", type: "app" }

and not `undefined`. I don't understand

Reply:
Si "app" and "type" sont dépréciés, pourquoi ne pas les supprimer et faire un BC ? A mon avis on peut se le permettre, le composant étant assez jeune et (chez nous) uniquement utilisé dans la home. Il est déjà assez compliqué comme ça...

Ca nous donnerait simplement `const appName = name || (get(appIconProps, 'app.name')) || ''`

Reply:
@doubleface 

here: https://github.com/cozy/cozy-ui/pull/1973/files#diff-fce2b54630424a9a413c69c08723d6e83ffc43e4ce8f8983c30325f9b97a4c7eR124 we pass to `<AppIcon ` `app={app}`

`app` comes from the props here:
https://github.com/cozy/cozy-ui/pull/1973/files#diff-fce2b54630424a9a413c69c08723d6e83ffc43e4ce8f8983c30325f9b97a4c7eR65

if you follow the sample you wrote here: https://github.com/cozy/cozy-ui/pull/1973/files#diff-7bbce49ffc2d1b4a041913caac7d27c2d1812a321158075dc1094df8978459b6R21

we don't pass any `app` props. So we're calling `<AppIcon` with `app={undefined}` no? And `AppIcon` need the slug in `app` props, right? 

At the end you're calling `<App app={undefined} type={type} slug={'myslug'} type={type} />` no? 

I'm wondering if we should not pass the AppIcon props directly to the SquareAppIcon without adding this `appIconProps` since there only `variant, IconContent` that is specific to SquareAppIcon. The result will be: 

```
const app = { name: "Test App", slug: "testapp", type: "app" }
<SquareAppIcon variant={XXX} IconComponent={CCC} {...app} />
```

I find it easier to understand. For instance this is how MUI works: Dialog can also get the props of Modal https://v4.mui.com/api/dialog/#inheritance

Reply:
> we don't pass any app props. So we're calling <AppIcon with app={undefined} no? And AppIcon need the slug in app props, right?

In this case app is undefined but appIconProps has an app props. Then AppIcon finally gets an app prop.

Reply:
> I'm wondering if we should not pass the AppIcon props directly to the SquareAppIcon without adding this appIconProps since there only variant, IconContent that is specific to SquareAppIcon. The result will be

That is what we got in the first place. But you made this [comment](https://github.com/cozy/cozy-ui/pull/1971/files#r749066601). Which was right I think, We will need the priority prop for the home application.

Reply:
Ok, non I understand. SquareAppIcon will get it's own props (variant, name, IconComponent) and pass the remaining props to AppIcon.